scripts/vehicle_node.py

```diff
@@ -201,78 +203,115 @@ def update_position(self, event):
+        no_rot = tf.transformations.quaternion_from_euler(0.0, 0.0, 0.0)
+        translation =  [self.joint1_length, 0.0, 0.0]
+        J1_T_STx = tf.transformations.quaternion_matrix(no_rot)
+        J1_T_STx[:3,3] = np.array(translation)
+
+        #Dynamic rotation of Joint2
+        dynamic_rot2 = tf.transformations.quaternion_from_euler(0.0, self.joint2_pitch, 0.0)
+        translation =  [0.0, 0.0, 0.0]
+        STx_T_J2 = tf.transformations.quaternion_matrix(dynamic_rot2)
+        STx_T_J2[:3,3] = np.array(translation)
+
+        #matrix multiplication to form the homogenous matrices
+        V_T_J1 = np.matmul(V_T_SRz, SRz_T_J1)
+        V_T_STx = np.matmul(V_T_J1, J1_T_STx)
+        V_T_J2 = np.matmul(V_T_STx, STx_T_J2)
+        rot_J1 = tf.transformations.quaternion_from_matrix(V_T_J1)
+        rot_J2 = tf.transformations.quaternion_from_matrix(V_T_J2)
+
+        #Assign pose information for publishing
+        self.joint1_pose.position.x = V_T_J1[0][3]
+        self.joint1_pose.position.y = V_T_J1[1][3]
+        self.joint1_pose.position.z = V_T_J1[2][3]
+        self.joint1_pose.orientation.x = rot_J1[0]
+        self.joint1_pose.orientation.y = rot_J1[1]
+        self.joint1_pose.orientation.z = rot_J1[2]
+        self.joint1_pose.orientation.w = rot_J1[3]
+
+        self.joint2_pose.position.x = V_T_J2[0][3]
+        self.joint2_pose.position.y = V_T_J2[1][3]
+        self.joint2_pose.position.z = V_T_J2[2][3]
+        self.joint2_pose.orientation.x = rot_J2[0]
+        self.joint2_pose.orientation.y = rot_J2[1]
+        self.joint2_pose.orientation.z = rot_J2[2]
+        self.joint2_pose.orientation.w = rot_J2[3]
+
+        #The ripper tip is a point in the J2's frame, this is based on the length of the ripper
+        ripper_point =  [self.ripper_length, 0.0, 0.0, 1.0]
+        map_T_J2 =  np.matmul(map_T_vehicle, V_T_J2)
+        tip =  np.matmul(map_T_J2, ripper_point)
+        ripper_tip = Point()
+        ripper_tip.x = tip[0]
+        ripper_tip.y = tip[1]
+        ripper_tip.z = tip[2]
+        points.append(ripper_tip)
+
+        #use the ripper's position as an index value to access the gaussian array
+        index_x = int(tip[0]/self.resolution)
+        index_y = int(tip[1]/self.resolution)
+
+        #Create a range of index values surrounding index_x and y
+        x = np.arange((index_x-1),(index_x+2), 1)
+        y = np.arange((index_y-1),(index_y+2), 1)
+        x1, y1 = np.meshgrid(x,y)
+        x2 = x1.ravel()
+        y2 = y1.ravel()
+
+        #First check if the index is within the gaussian array, if it is, then check if the tip of
+        #the ripper is beneath the soil, if it is, then remove the soil above the tip and disperse
+        #it to the surrounding cells, provided those cells are also within the gaussian array
+
+        if (0 <= index_x <= self.gaussian_array.shape[0]) and (0 <= index_y <= self.gaussian_array.shape[1]):
+            if (self.gaussian_array[index_x][index_y] >= tip[2]):
```

Comment:
Shouldn't this line be (self.gaussian_array[index_x][index_y] > tip[2])? this branch of logic would not really do anything if they were equal.